### ISSUE TITLE
[$75] Web - Expensify Card - Inconsistent behavior of phone number validation for shipping details

### ISSUE THREAD

**Issue by IuliiaHerets** [Expensify/App#51491]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 9.0.53-0
**Reproducible in staging?:** Y
**Reproducible in production?:** Y
**Email or phone of affected tester (no customers):** nathanmulugetatesting+1943@gmail.com
**Issue reported by:** Applause Internal Team

## Action Performed:
Pre-conditions:
- Workspace is connected with a bank account
- Workspace has a member invited
- Workspace has enabled Expensify card

1. Issue a physical card for the member
2. Sign in as the member
3. Go to the workspace chat and click on add shipping detail page
4. Go through the flow until phone number input page
5. Input this: 8005555555 phone number and click enter
6. Observe the behavior of the error when 1 is prepended
7. Go to settings > Wallet > click on the assigned card under the Assigned cards section
8. Go through the flow until you get to the phone number input page
9. Input 8005555555 and click enter
10. Observe the behavior of the error when you prepend 1
11. Prepend "+" again and observe the behavior of the error

## Expected Result:
The phone number validation works the same on both pages

## Actual Result:
The phone number validation works inconsistently, when going through the form from the workspace chat page prepending "1" only works fine but when going through the form from the wallet page user has to additionally prepend "+" before "1" in order for the phone number to be accepted.

## Workaround:
Unknown

## Platforms:

- [ ] Android: Standalone
- [ ] Android: HybridApp
- [ ] Android: mWeb Chrome
- [ ] iOS: Standalone
- [ ] iOS: HybridApp
- [ ] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

https://github.com/user-attachments/assets/c567cbf0-0c41-4fd3-8c8d-74c3377424c6

  
</details>

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~021850956027322769046</li>
        <li>Upwork Job ID: 1850956027322769046</li>
        <li>Last Price Increase: 2025-02-17</li>
<li>Automatic offers: </li>
<ul>
<li>NJ-2020 | Contributor | 104756183</li>
</ul></li>
    </ul>
</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @sobitneupane</details>

**Comment by IuliiaHerets** [comment]
We think that this bug might be related to #wave-collect - Release 2

**Comment by IuliiaHerets** [comment]
@garrettmknight  FYI I haven't added the External label as I wasn't 100% sure about this issue. Please take a look and add the label if you agree it's a bug and can be handled by external contributors

**Comment by sobitneupane** [comment]
Thanks for the proposal, @NJ-2020 and @twilight2294 

Were you able to reproduce the issue?

There are several components in the app that use phone numbers, so I think it would be good to review all of them to ensure consistent behavior.

**Comment by sobitneupane** [comment]
Thanks for the proposal @NJ-2020 

[Proposal](https://github.com/Expensify/App/issues/51491#issuecomment-2439042648) from @NJ-2020 looks good to me.

🎀 👀 🎀 C+ reviewed

**Comment by chiragsalian** [comment]
Proposal LGTM, feel free to create the PR @NJ-2020

**Comment by sobitneupane** [comment]
Waiting on @NJ-2020 to make a few changes in the PR.

**Comment by jasperhuangg** [comment]
We are reverting this PR

**Comment by sobitneupane** [comment]
@NJ-2020 You are right, the issue I mentioned is probably not related to the PR. However, you pointed out [here](https://github.com/Expensify/App/issues/54487#issuecomment-2560603735) that the original issue is coming from the PR. Were you able to reproduce the issue?

**Comment by sobitneupane** [comment]
@chiragsalian [The PR](https://github.com/Expensify/App/pull/52082) was reverted to resolve [this issue](https://github.com/Expensify/App/issues/54487). We could not reproduce the exact issue on the PR, but a [similar issue](https://github.com/Expensify/App/issues/54614) is reproducible even after the revert.

I think we should recreate [the PR](https://github.com/Expensify/App/pull/52082) and request the QA team to test it thoroughly to try and reproduce the issue before merging it.

**Comment by sobitneupane** [comment]
I see a thumbs-up in the above comment by @chiragsalian. I believe we can proceed with your PR, @NJ-2020 and request QA team to try and reproduce [this issue](https://github.com/Expensify/App/issues/54487).

**Comment by sobitneupane** [comment]
Issue is not reproducible any longer. 

https://github.com/user-attachments/assets/6d80145a-b9a4-481a-82f2-cc654b70bc47

**Comment by garrettmknight** [comment]
I'll still pay out for this, but I'm lowering again to $75 since it's been on the go since November and we solved elsewhere in the meantime, on top of causing a regression/.

**Payment Summary:**
- Contributor: @NJ-2020 $75 paid via Upwork
- Contributor+: @sobitneupane $75 via NewDot

**Comment by garrettmknight** [comment]
@sobitneupane request when you're ready!

**Comment by JmillsExpensify** [comment]
Payment Summary:

Contributor: @NJ-2020 $75 paid via Upwork
Contributor+: @sobitneupane $75 via NewDot
